### PR TITLE
Make LIF cells LIF cells again.

### PR DESCRIPTION
**Pull request by thorstenhater**
lif

- Add (again) a reset voltage E_R such that V_m = E_R in the refractory period.
- Make V_m decay towards resting potential E_L instead of 0.
- Adjust docs and tests accordingly.
- Double check this against literature.

### CODE REVIEW

**Review by brenthuisman — CHANGES_REQUESTED** [review]
One tiny thing!

**Comment by thorstenhater** [comment]
What would that be? :) There are no comments?

File: doc/concepts/lif_cell.rst [hunk 1]
```diff
@@ -21,20 +23,19 @@ mechanisms.
 
 The LIF cell's time dynamics are this:
 
-0. :math:`U_\mathrm{m}(0) = E_\mathrm{L}`
-1. If the cell is in its refractory state :math:`U_\mathrm{m}(t) = E_\mathrm{L}`
-2. Otherwise :math:`U'_\mathrm{m}(t) = \sum\limits_\mathrm{spike} w_\mathrm{spike} \cdot\delta(t - t_\mathrm{spike}) -\frac{1}{\tau_\mathrm{m}}U_\mathrm{m}(t)`
+0. :math:`U_\mathrm{m}(0) = V_\mathrm{0}`,
+1. If the cell is in its refractory state :math:`U_\mathrm{m}(t) = E_\mathrm{R}`
+2. Otherwise :math:`U'_\mathrm{m}(t) = \sum\limits_\mathrm{spike} w_\mathrm{spike} \cdot\delta(t - t_\mathrm{spike}) - \frac{1}{\tau_\mathrm{m}}\left(U_\mathrm{m}(t)) - E_\mathrm{L}\right)`
```
Comment (brenthuisman) [inline doc/concepts/lif_cell.rst:28]: `w_\mathrm{spike}` is not defined above.

Comment (thorstenhater) [reply doc/concepts/lif_cell.rst:28]: Yes, it's not part of the cell, but it should still be defined somewhere.

Comment (thorstenhater) [reply doc/concepts/lif_cell.rst]: There, it is now explained. I added C++ API docs for LIF as well.

File: doc/concepts/lif_cell.rst [hunk 2]
```diff
@@ -21,20 +23,21 @@ mechanisms.
...
 .. figure:: ../images/lif.svg
-    :width: 400
+    :width: 600
     :align: center
 
     Plot of the potential over time for a LIF cell.
 
-
 API
 ---
```
Comment (brenthuisman) [inline doc/concepts/lif_cell.rst:43]: Not sure why this link does not resolve, see: https://arbor--2026.org.readthedocs.build/en/2026/concepts/lif_cell.html

**Review by brenthuisman — APPROVED** [review]
Hurray!

**Pull request merged**